### ISSUE TITLE
domain model attribute of type id problem

### ISSUE THREAD

**Issue by cgreinke** [nativelogix/xquerrail2.framework#40]
Upon saving a map of a model, IDs for elements are being saved incorrectly. IDs are duplicated instead of staying unique. This seems to occur when updating an existing model with data in a different order than what was there originally.

In domain model definition file:
```xml
<domain ...>
    <model name="user" ...>
    ...
    <container name="listSubscriptions" label="List Subscriptions">
        <element name="listSubscription" label="List Subscription" type="listSubscription" occurrence="*">...</element>
        ...
    </container>
    ...
    </model>

    <model name="listSubscription" label="List Subscription" persistence="abstract">
        <attribute name="id" type="id"/>
        <element name="list" label="List" type="reference" reference="model:list:reference">...</element>
        ...
    </model>
    ...
</domain>
```
When passing the following map:
```xml
<map:entry key="listSubscriptions.listSubscription">
    <map:value>
        <listSubscription xsi:type="listSubscription" id="nd2d660accf3b978f">
            <list ref-type="model" ref-uuid="01580dd5-34bb-8d29-0bcb-9850280c072891fa53b5" ref-id="01580dd5-34bb-8d29-0bcb-9850280c072891fa53b5" ref="list">another value</list>
        </listSubscription>
    </map:value>
    <map:value>
        <listSubscription xsi:type="listSubscription" id="n2eddb954951b0af1">
            <list ref-type="model" ref-uuid="a2d05688-b690-815b-d027-c98746e4ccdfca38c650" ref-id="a2d05688-b690-815b-d027-c98746e4ccdfca38c650" ref="list">some value</list>
        </listSubscription>
    </map:value>
</map:entry>
```
To update the following model element:

```xml
<listSubscriptions>
    <listSubscription xsi:type="listSubscription" id="n2eddb954951b0af1">
        <list ref-type="model" ref-uuid="a2d05688-b690-815b-d027-c98746e4ccdfca38c650" ref-id="a2d05688-b690-815b-d027-c98746e4ccdfca38c650" ref="list">some value</list>
    </listSubscription>
</listSubscriptions>
```

I'll get the following result:

```xml
<listSubscriptions>

    <listSubscription xsi:type="listSubscription" id="nd2d660accf3b978f">
        <list ref-type="model" ref-uuid="01580dd5-34bb-8d29-0bcb-9850280c072891fa53b5" ref-id="01580dd5-34bb-8d29-0bcb-9850280c072891fa53b5" ref="list">another value</list>
    </listSubscription>

    <listSubscription xsi:type="listSubscription" id="nd2d660accf3b978f">
        <list ref-type="model" ref-uuid="a2d05688-b690-815b-d027-c98746e4ccdfca38c650" ref-id="a2d05688-b690-815b-d027-c98746e4ccdfca38c650" ref="list">some value</list>
    </listSubscription>

</listSubscriptions>
```

Instead of the desired result:
```xml
<listSubscriptions>

    <listSubscription xsi:type="listSubscription" id="nd2d660accf3b978f">
        <list ref-type="model" ref-uuid="01580dd5-34bb-8d29-0bcb-9850280c072891fa53b5" ref-id="01580dd5-34bb-8d29-0bcb-9850280c072891fa53b5" ref="list">another value</list>
    </listSubscription>

    <listSubscription xsi:type="listSubscription" id="n2eddb954951b0af1">
        <list ref-type="model" ref-uuid="a2d05688-b690-815b-d027-c98746e4ccdfca38c650" ref-id="a2d05688-b690-815b-d027-c98746e4ccdfca38c650" ref="list">some value</list>
    </listSubscription>

</listSubscriptions>
```

Notice that the ids of the listSubscription elements are the same in the actual result but different in the desired result.

If I switched the order of map values in the map, I got the desired result.

**Comment by rlouapre** [comment]
Please take a look at append [1] and prepend [2] unit tests and associated domain model [3].  
These tests are passing. Do they match your requirements?

[1] - https://github.com/nativelogix/xquerrail2.framework/blob/master/src/test/_framework/base/base-model-test/base-model-crud-test.xqy#L889
[2] - https://github.com/nativelogix/xquerrail2.framework/blob/master/src/test/_framework/base/base-model-test/base-model-crud-test.xqy#L923
[3] - https://github.com/nativelogix/xquerrail2.framework/blob/master/src/test/_framework/base/base-model-test/app-test/domains/content-domain.xml#L176

**Comment by cgreinke** [comment]
I think you meant to link to

-  model-append-new-item-container-element-with-id-test
-  model-prepend-new-item-container-element-with-id-test

If this is correct, then I guess that I would have the following comments:

1. Given that, in my example, ids were being duplicated, I would add a check to both test cases to assert that both entries have different id values. This, along with your existing check that the preexisting abstract instance's id did not change, would satisfy my requirements. 

2. This is nit picky, but I would probably ensure that ```$new-abstract-name``` doesn't equal ```$abstract-value-name``` before updating ```$instance19```.  Because I don't know what you seeded ```$instance19``` with, and the fact that random does not equal unique, I would add this check. This with your existing check that the values didn't change would be good enough.

**Comment by rlouapre** [comment]
I would suggest to upgrade to [v0.0.4](https://github.com/nativelogix/xquerrail2.framework/releases/tag/v0.0.4)

**Comment by cgreinke** [comment]
Sweet. Thanks!

**Comment by rlouapre** [comment]
I found few issues in [v0.0.4](https://github.com/nativelogix/xquerrail2.framework/releases/tag/v0.0.4). I would suggest to upgrade to [v0.0.5](https://github.com/nativelogix/xquerrail2.framework/releases/tag/v0.0.5)

**Comment by rlouapre** [comment]
Closing this issue. Has been fixed since [v0.0.4](https://github.com/nativelogix/xquerrail2.framework/releases/tag/v0.0.4) but recommend to upgrade to [v0.0.10](https://github.com/nativelogix/xquerrail2.framework/releases/tag/v0.0.10)
